src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -59,7 +57,7 @@ public static void ReplaceFile(string sourceFullPath, string destFullPath, strin
             }
         }
 
-        public static void CreateDirectory(string fullPath)
+        public static void CreateDirectory(string fullPath, DirectorySecurity dirSecurity = null)
```

Comment:
Have this take the security descriptor instead, then there will be no dependencies.

``` C#
namespace System.IO
{
    public static class FileSystemAclExtensions
    {
        public static void Create(
            this DirectoryInfo directoryInfo,
            DirectorySecurity directorySecurity)
        {
            // Parameter validation here

            FileSystem.CreateDirectory(directoryInfo.FullName, directorySecurity.GetSecurityDescriptorBinaryForm());
        }
    }

    internal static partial class FileSystem
    {
        public unsafe static void CreateDirectory(string fullPath, byte[] securityDescriptor = null)
        {
            // Existing goo

            fixed (byte* b = securityDescriptor)
            {
                var securityAttributes = new Interop.Kernel32.SECURITY_ATTRIBUTES
                {
                    nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
                    lpSecurityDescriptor = (IntPtr)b
                };

                // Existing create directory loop here, eg
                // while()
                // {
                //  r = Interop.Kernel32.CreateDirectory(name, ref secAttrs);
                // }
            }
        }
    }
}
```